tests/functional/views/basic_lti_launch_test.py

```diff
@@ -329,15 +330,27 @@ def canvas_url_lti_params(self, lti_params, sign_lti_params):
         )
 
     @pytest.fixture
-    def canvas_file_legacy_speedgrader_lti_params(
-        self, canvas_file_lti_params, sign_lti_params
+    def canvas_file_legacy_speedgrader_launch_params(
+        self, canvas_file_launch_params, sign_lti_params
     ):
         # Legacy SpeedGrader launches will send the wrong resource_link_id
         # on the POST params and not include the right one on the query params.
-        canvas_file_lti_params["resource_link_id"] = canvas_file_lti_params[
-            "context_id"
-        ]
-        return sign_lti_params(canvas_file_lti_params)
+        _, post_params = canvas_file_launch_params
+        return {"learner_canvas_user_id": "USER_ID"}, sign_lti_params(
+            dict(
+                post_params,
+                resource_link_id=post_params["context_id"],
+            )
+        )
+
+    @pytest.fixture
+    def canvas_file_fixed_speedgrader_launch_params(self, canvas_file_launch_params):
+        _, post_params = canvas_file_launch_params
+        return {
+            "learner_canvas_user_id": "USER_ID",
+            # Fixed (non-legacy) SpeedGrader launches include resource_link_id in the URL params
```

Comment:
Ok so a "fixed" SpeedGrader launch means a non-legacy one. I think maybe we should call them "SpeedGrader launches" and "legacy SpeedGrader launches"